doc/guides/optimizations/footprint.rst

```diff
@@ -0,0 +1,55 @@
+.. _footprint:
+
+Optimizing for Footprint
+########################
+
+Stack Sizes
+***********
+
+Stack sizes of various system threads are sepcified generously to allow for
+usage in different scenarios on as many supported platforms as possible. You
+should start the optimization process by reviewing all stack sizes and adjusting
+them for your application:
+
+CONFIG_ISR_STACK_SIZE
+  Set to 2048 by default
+
+CONFIG_MAIN_STACK_SIZE
+  Set to 1024 by default
+
+CONFIG_IDLE_STACK_SIZE
+  Set to 320 by default
+
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE
+  Set to 1024 by default
+
+
+Unused Peripherals
+******************
+
+Some peripherals are enabled by default, if you for example not using GPIO, you
+should disable it in your project configuration:
+
+        CONFIG_GPIO=n
+        CONFIG_SPI=n
+
+Various Debug/Informational Options
+***********************************
+
+The following options are enabled by default to provide more information about
+the running application and to provide means for debugging and error handling:
+
+CONFIG_BOOT_BANNER
+  This option can be disabled to save a few bytes.
+
+CONFIG_DEBUG
+  This option can be disabled for production builds
```

Comment:
For production systems logging can be disabled, or at least it could be mentioned here how to set the global max log level.